File: tmdb-core/src/main/java/dev/reuss/tmdb/value/id/TvEpisodeGroupId.java
package dev.reuss.tmdb.value.id;

/**
 * TMDB TV episode group id.
 *
 * @param value TMDB TV episode group id
 */
public record TvEpisodeGroupId(String value) implements TmdbResourceId {

    public TvEpisodeGroupId {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("TV episode group id must not be blank");
        }

        value = value.trim();
    }

    public static TvEpisodeGroupId of(String value) {
        return new TvEpisodeGroupId(value);
    }

    @Override
    public String asString() {
        return value;
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/value/id/TvEpisodeGroupId.kt
package dev.reuss.tmdb.value.id

/**
 * TMDB TV episode group id.
 */
class TvEpisodeGroupId private constructor(
    value: String,
) : StringTmdbResourceId(value, "TV episode group id") {
    companion object {
        /**
         * Creates an identifier from a non-blank value.
         *
         * Leading and trailing whitespace is removed.
         *
         * @throws IllegalArgumentException if [value] is blank
         */
        @JvmStatic
        fun of(value: String): TvEpisodeGroupId = TvEpisodeGroupId(value)
    }
}
